Mage/src/main/java/mage/game/permanent/PermanentImpl.java

```diff
@@ -1351,7 +1351,17 @@ public boolean cantBeAttachedBy(MageObject attachment, Ability source, Game game
                 return !ability.getDoesntRemoveControlled() || isControlledBy(game.getControllerId(attachment.getId()));
             }
         }
-        return game.getContinuousEffects().preventedByRuleModification(new StayAttachedEvent(this.getId(), attachment.getId(), source), null, game, silentMode);
+
+        boolean canAttach = true;
+        Permanent attachmentPermanent = game.getPermanent(attachment.getId());
+        if (attachment.hasSubtype(SubType.AURA, game)
+                && attachmentPermanent != null
+                && attachmentPermanent.getSpellAbility() != null
+                && !attachmentPermanent.getSpellAbility().getTargets().isEmpty()) {
+            canAttach = game.getPermanent(attachment.getId()).getSpellAbility().getTargets().get(0).getFilter().match(this, game);
```

Comment:
Should probably use the 4 argument match(..)
Or else stuff like "Enchant creature you control" won't work properly.
Could you add a test on Codsworth trying to attach [[Captured by the Consulate]] to one of your creature (it shouldn't move)

Reply:
Edit: Uhm you might not have easy access right now to the 4 argument match as it is only defined for FilterPermanent, FilterCard, FilterPlayer, etc...

Should `Filter` be changed to have 4 argument match defined? not sure.

Reply:
> Could you add a test on Codsworth trying to attach [[Captured by the Consulate]] to one of your creature (it shouldn't move)

Having written this test now, I can confirm it is currently failing. I'm not sure what you mean about how to account for this case?

Reply:
I added the 4 argument match change you suggested to `Permanent::cantBeAttachedBy` and can confirm that the previously failing test now passes. However, adding the checks for Player / Card to Permanent would be redundant as if we are calling the Permanent's cantBeAttachedBy method it's because we're trying to attach to a permanent specifically. The issue being that Players and Cards (and I suppose any other object, auras don't discriminate) don't have a `cantBeAttachedBy` equivalent. I've written two tests for the cases of trying to use Codsworth to move auras that enchant players / cards to permanents which of course currently has the auras move illegally, and then sends them to the graveyard during state based actions. Am I missing something or does this need to be fixed via adding `cantBeAttachedBy` equivalents to these objects (Player and Card)?

Reply:
It would make sense to have `cantBeAttached` for Cards and Players, that for equipment/fortification always return true (as you can only attach an equipement to creatures and a fortification to a land), and for auras, checks that the target is of the proper kind, and checks its prerequisite.

Side note: are curses properly falling off right now when they are named by Runed Halo?

Reply:
> Side note: are Curses properly falling off right now when they are named by Runed Halo?

Yes they are, but will add a test for that to this PR for future-proofing.

Reply:
Alright, thanks for testing.
Those Attachment related methods sure do a lot in a various different places of the codebase.